### DISCUSSION TITLE
11.1.0-beta1

### DISCUSSION THREAD

**Discussion by maxkatz6** [AvaloniaUI/Avalonia#14922] [General]
**Full Changelog**: https://github.com/AvaloniaUI/Avalonia/commits/11.1.0-beta1
**Full API Diff**: https://gist.github.com/maxkatz6/a7df42607cc7187b94e2e693ccf15f6d

Please install latest [dotnet templates](https://github.com/AvaloniaUI/avalonia-dotnet-templates).
Most of new features are still pending updated documentation.
You can follow progress on this issue - https://github.com/AvaloniaUI/avalonia-docs/issues/384.

## New features
- Binding markup extension was refactored, and now supports UpdateSourceTrigger!
- Tizen backend with Tizen TV support.
- XAML warnings support, starting with Obsolete and TemplatePart warnings. Configurable via csproj and editorconfig. XAML error reporting is now also improved, not being limited by a single error at a time.
- New top-level and application services: ILauncher, IInputPane, IActivatableLifetime.
- XY focus with arrow keys and remote DPad is now available.
- Global exception handling via Dispatcher UnhandledException and UnhandledExceptionFilter.

## New and extended controls:
- HyperlinkButton is now available.
- MenuItem now supports ToggleType property, including CheckBox and Radio types with radio groups.
- Border now includes a new BackgroundSizing property, allowing customizing transparent border brushes rendering.
- Many interactive controls were updated to be usable with TV remote control, where only DPad buttons and Space (Select) + Escape (Back) are available.
- NumericUpDown and AutoCompleteBox: MaxLength, InnerLeftContent, and InnerRightContent.
- WindowNotificationManager was refactored and now can be embedded at any level of the controls tree, not being hardcoded to the top level.
- RadialGradientBrush now has separated `RadiusX` and `RadiusY` properties that allow absolute and relative values. Old `Radius` property is now obsolete.

### Windows
- DirectComposition rendering mode is now supported for older Windows versions (8.1+)
- It's now possible to add `WndProc` and `WindowStyles` callbacks via `Win32Properties` API.

### macOS
- `TrayIcon` now includes `MacOSProperties.IsTemplateIcon` attached property, allowing monochrome icons consistent with macOS system icons

### Linux
- It's now possible to set `NetWmWindowType` and `WmClass` on any window via `X11Properties` attached properties.
- Avalonia now respects XGD accent color spec.

### Browser
- `StaticWebAssets` SDK is now supported. It brings support for building in .NET Brotli compression and caching. To use this SDK, please recreate the browser project from the new template using the 11.1.0-beta1 version.
- WebGL2 is now enabled by default.

### iOS
- Metal rendering mode was implemented and now can be enabled via iOSPlatformOptions.
- Physical keyboard and mouse input handling now work.
- **tvOS support was added**.

### Android
- Android TV support was added

## **Breaking Changes**
- iOS and Android backends now require .NET 8 or newer, as per .NET mobile workloads support timelines.
- Browser backend now also requires .NET 8 or newer, but also has changes to its project structure. We recommend using newer Avalonia templates as a reference, but a migration guide specifically for the browser is also planned.

<hr /><em>This discussion was created from the release <a href='https://github.com/AvaloniaUI/Avalonia/releases/tag/11.1.0-beta1'>11.1.0-beta1</a>.</em>

**Comment by qq908382818** [comment]
synchronize Avalonia.Templates to V11.1 thanks!

**Reply by maxkatz6** [reply]
Sure, we plan it for when 11.1 gets stable release (not beta) :)

**Reply by qq908382818** [reply]
Why are the intelligent prompts not available or inaccurate when using VS2022?

**Reply by timunie** [reply]
check if your extensions are up to data. Make sure you have built the project once in a while

**Reply by AlexeyTliss** [reply]
WndProck access!
NetWmWindowType access!
Checkboxes/radiobuttons in tray menus! 
XAML warnings and UpdateSourceTrigger!
Awesome!!

**Comment by timunie** [comment]
If you want us to help getting 11.1 as stable as possible, would be very useful if you can re-visit your opened issues and double-check if 11.1-beta1 solves or changes anything for you.

**Reply by llfab** [reply]
Seems to solve #9659 (tried in our apps)

**Comment by robloo** [comment]
Is it worth mentioning `BackgroundSizing`? It is low level but control libraries and people using transparent borders need to be aware of it.

**Reply by maxkatz6** [reply]
I think it is. I tried to mention as many API changes. And we have this annoying default behavior, about which I wish I knew earlier developing apps with Avalonia.

**Reply by robloo** [reply]
Oh, I see it. Sorry I missed it before.

Concerning the default: perhaps we could change it to the same as WinUI instead of CenterBorder? It could be done in the control templates. It might not actually be a huge breaking change if done in the themes.

**Reply by maxkatz6** [reply]
Personally, I would vote for changing, yes. But not for 11.1.

**Comment by qq908382818** [comment]
I want to get the server IP and client IP. Can I get them in the  web type avalonia?

**Reply by maxkatz6** [reply]
Web type avalonia is just a web app in the browser. There is no server. Unless you made one yourself. 
Any app also doesn't know its own client IP.

**Comment by danipen** [comment]
Hey team, any updates on when the official release of version `11.1.0` will be available? Thanks!

**Reply by timunie** [reply]
As soon as it's stable enough. I think there will be a beta2 at minimum.

**Comment by Sigrec** [comment]
I'm not seeing the **11.1.0-beta1** in the nightly package source, this is my source path **https://nuget-feed-nightly.avaloniaui.net/v3/index.json** am I missing something?

**Reply by timunie** [reply]
Should be on main nuget feed. Make sure preview checkmark is active.
